packages/vite-plugin-cloudflare/src/node-compat.ts

```diff
@@ -0,0 +1,92 @@
+import assert from 'node:assert';
+import * as path from 'node:path';
+import { getNodeCompat } from 'miniflare';
+import { cloudflare, env, nodeless } from 'unenv';
+
+/**
+ * Get Miniflare modules and code snippets that are needed if running in Nodejs compat mode.
+ */
+export function getNodeCompatModules(
+	compatibilityDate: string | undefined,
+	compatibilityFlags: string[] = [],
+	modulesRoot: string,
+) {
+	if (getNodeCompat(compatibilityDate, compatibilityFlags).mode !== 'v2') {
+		return { nodeModules: [], nodeWrappers: [] };
+	}
+
+	const { alias, inject } = env(nodeless, cloudflare);
+
+	const nodeWrappers = Object.entries(inject).map(
+		([globalName, globalInject]) =>
+			getGlobalModuleContents(globalName, globalInject),
+	);
+
+	const nodeModules = Object.entries(alias)
+		.map(deAliasing(alias))
+		.filter(removeProcess)
+		.filter(removeIdentities)
+		.map(([importSpecifier, modulePath]) => {
+			modulePath = modulePath.replace(/proxy-cjs$/, 'proxy');
+			return {
+				type: 'ESModule',
+				path: path.join(modulesRoot, importSpecifier),
+				contents: `export { default } from '${modulePath}';`,
+			} as const;
+		});
+
+	return { nodeModules, nodeWrappers };
+}
+
+/**
+ * Get the import statement and export name to be used for the given global inject setting.
+ */
+function getGlobalModuleContents(
+	globalName: string,
+	globalInject: string | string[],
+) {
+	if (typeof globalInject === 'string') {
+		const moduleSpecifier = globalInject;
+		// the mapping is a simple string, indicating a default export, so the string is just the module specifier.
+		return `import var_${globalName} from "${moduleSpecifier}";globalThis.${globalName} = var_${globalName};`;
+	}
+
+	// the mapping is a 2 item tuple, indicating a named export, made up of a module specifier and an export name.
+	const [moduleSpecifier, exportName] = globalInject;
+	return `import var_${globalName} from "${moduleSpecifier}";\nglobalThis.${globalName} = var_${globalName}.${exportName};`;
+}
+
+function deAliasing(alias: Record<string, string>) {
+	return (
+		[importSpecifier, modulePath]: [string, string],
+		index: number,
+	): [string, string] => {
+		const breadcrumbs: (string | undefined)[] = [importSpecifier];
```

Comment:
is this type correct/necessary?
```suggestion
		const breadcrumbs = [importSpecifier];
```